compiler+runtime/src/cpp/jank/analyze/processor.cpp

```diff
@@ -3662,8 +3662,28 @@ namespace jank::analyze
     }
 
     auto const string_obj(l->data.rest().first().unwrap());
+
+    /* Only apply guards if the argument is a string literal */
+    runtime::object_ref guarded_string_obj{ string_obj };
+    if(string_obj->type == runtime::object_type::persistent_string)
```

Comment:
We already do a check below to ensure that `cpp/raw` only receives a string. If we give it anything else, we have an error. We should just update the string in the `primitive_literal` expression that we get back later in this function. We have it bound as `auto const obj`.